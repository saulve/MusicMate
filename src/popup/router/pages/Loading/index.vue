<template>
  <spinner/>
</template>

<script>
import Spinner from "vue-simple-spinner";

export default {
  components: {
    spinner: Spinner
  },
  data() {
    return {};
  },
  mounted() {
    /**
     *
     * I takes a bit of time for the background store instance
     * to be copied into the newly instantiated popup store
     * therefore we're watching for the change and only proceed
     * when both states (background and popup) are in sync
     */
    this.$store.watch(
      state => state.timer,
      () => {
        if ( this.$store.getters.isLoggedIn ) {
          this.$router.push( "/main" );
        } else {
          this.$router.push( "/login" );
        }
      }
    );
  }
};
</script>
