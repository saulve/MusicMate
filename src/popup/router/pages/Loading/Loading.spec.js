import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import Spinner from "vue-simple-spinner";
import Loading from "./index.vue";

const localVue = createLocalVue();

localVue.use( Vuex );

describe( "Loading Page", () => {
  let store;

  beforeEach( () => {
    store = new Vuex.Store();
  });

  it( "renders component", () => {
    const wrapper = shallowMount( Loading, { store, localVue });
    expect( wrapper.element ).toMatchSnapshot();
    expect( wrapper.contains( Spinner ) ).toBe( true );
  });
});
