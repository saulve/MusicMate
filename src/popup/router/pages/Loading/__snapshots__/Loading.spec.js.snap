// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Page renders component 1`] = `
<spinner-stub
  fontsize="13"
  linebgcolor="#eee"
  linefgcolor="#2196f3"
  linesize="3"
  message=""
  size="32"
  spacing="4"
  speed="0.8"
  textfgcolor="#555"
/>
`;
